server/src/main/java/io/crate/types/ResultSetParser.java

```diff
@@ -80,11 +80,12 @@ public static Object getObject(ResultSet resultSet, int i, String typeName) thro
                 return PgOidVectorType.listFromOidVectorString(textval);
             }
             case "char":
-                String strValue = resultSet.getString(columnIndex);
-                if (strValue == null) {
+                // It's not a single byte type (CharType) as char is not surrounded by quotes.
+                var str = resultSet.getString(columnIndex);
+                if (str == null) {
                     return null;
```

Comment:
flaky tests - `testAnyOnArrayLiteral` with `UseJdbc = 1 `fails consistently.

We have to differentiate here between `char` and `"char"`. Problem is that driver uses OID to resolve type name and OID is correctly assigned to 18 in CharType. In postgres pg_catalog.pg_type doesn't contain special entry for "char".